app/actions/combat.py

```diff
@@ -0,0 +1,27 @@
+from app.actions.preparations import make_pairs
+from app.people.knights import Knight
+
+
+def fight(knights: dict) -> dict:
+    result = {}
+
+    pairs = make_pairs(knights)
+
+    for pair in pairs:
+        first_knight = pair[0]
+        second_knight = pair[1]
+
+        first_knight.hit(second_knight)
+        second_knight.hit(first_knight)
+        check_hp(first_knight)
+        check_hp(second_knight)
```

Comment:
u can implement logic for both of knights - keep ur code DRY

Reply:
I modified my check_hp function and make one line